src/Haskell/Ide/Engine/Transport/JsonHttp.hs

```diff
@@ -27,29 +27,45 @@ import qualified Data.Map as Map
 import           Data.Maybe
 import           Data.Proxy
 import           Data.Singletons.Prelude hiding ((:>))
+import           Data.Swagger (Swagger)
 import qualified Data.Text as T
-import           GHC.Generics
 import           GHC.TypeLits
 import           Haskell.Ide.Engine.PluginDescriptor
 import           Haskell.Ide.Engine.Transport.JsonHttp.Undecidable
 import           Haskell.Ide.Engine.Types
 import           Network.Wai
-import           Network.Wai.Handler.Warp
+import           Network.Wai.Handler.Warp as Warp
+import qualified Paths_haskell_ide_engine as Meta
 import           Servant
 import           Servant.Server.Internal
 
--- | A greet message data type
-newtype Greet = Greet { _msg :: T.Text }
-  deriving (Generic, Show)
+{-# ANN module ("HLint: ignore Eta reduce" :: String) #-}
 
-instance FromJSON Greet
-instance ToJSON Greet where
-  toJSON = genericToJSON defaultOptions
 
 newtype TaggedMap (tags :: [ParamDescType]) = TaggedMap ParamMap deriving (Monoid)
 
 instance TaggedMapParser tags => FromJSON (TaggedMap tags) where
-  parseJSON (Object v) = fmap (TaggedMap . Map.fromList) (parseTaggedMap (Proxy :: Proxy tags) v)
+  parseJSON (Object v) = do
+    {-
+       The initial version of the hie API expected parameters in the POST body of the form
+
+         {"command":{"text": "version"},"plugin":{"text": "base"}}
+
+       The swagger version can only accept one parameter in the body, so this must be wrapped as
+
+         { "params:
```

Comment:
Once more a victim of pressing ", getting 2, deleting one and they both disappear.